src/openforms/forms/models/form_definition.py

```diff
@@ -114,6 +118,17 @@ def copy(self):
             else ""
         )
         copy.slug = _("{slug}-copy").format(slug=self.slug)
+
+        # truncate name and internal name if needed
+        copy.name = truncate_str_if_needed(
+            self.name, copy.name, self.get_charfield_max_length("name")
+        )
+        copy.internal_name = truncate_str_if_needed(
+            self.internal_name,
+            copy.internal_name,
+            self.get_charfield_max_length("internal_name"),
+        )
```

Comment:
I am thinking: the slug also has a max length of 100 char, and since it is derived (usually) from the name, it also has a chance of being too long :thinking:

Reply:
Yes, it was my first thought but then didn't touch it as it can be too long only if the user explicitly defines something like 90 chars as a slug. But yeah, just to be safe, I will add the slug as well.